syne_tune/tuner_callback.py

```diff
@@ -101,3 +101,17 @@ def on_tuning_sleep(self, sleep_time: float):
         :param sleep_time: Time (in secs) for which tuner has just slept
         """
         pass
+
+    def on_start_trial(self, trial: Trial):
```

Comment:
These methods were originally missing. They are also needed in the 2nd part of the PR